library/src/test/java/kwasm/format/text/TableTypeTest.kt

```diff
@@ -0,0 +1,161 @@
+
+import com.google.common.truth.Truth.assertThat
+import kwasm.format.ParseContext
+import kwasm.format.ParseException
+import org.junit.Assert.assertThrows
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+@UseExperimental(ExperimentalUnsignedTypes::class)
+class TableTypeTest {
+
+    private val context = ParseContext("TokenizerTest.wasm", 1, 1)
+    private val tokenizer = Tokenizer()
+
+    @Test
+    fun parseSingleNumber_withoutFuncref_throwsParseException() {
+        val expectedMin = 123456.toUInt()
+        val tokens = tokenizer.tokenize("$expectedMin", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Too few arguments")
+    }
+
+    @Test
+    fun parseSingleNumber_withoutNonFuncrefLiteral_throwsParseException() {
+        val expectedMin = 123456.toUInt()
+        val tokens = tokenizer.tokenize("$expectedMin notFuncref", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Expected 'funcref'")
+    }
+
+    @Test
+    fun parseSingleNumber_setMinToCorrectValue() {
+        val expectedMin = 123456.toUInt()
+        val expectedMax = UInt.MAX_VALUE
+        val tokens = tokenizer.tokenize("$expectedMin funcref", context)
+        val parseResult = tokens.parseTableType(0)
+        assertThat(parseResult.astNode.limits.min).isEqualTo(expectedMin)
+        assertThat(parseResult.astNode.limits.max).isEqualTo(expectedMax)
+        assertThat(parseResult.parseLength).isEqualTo(2)
+    }
+
+    @Test
+    fun parseMaxVal_setMinToMaxVal() {
+        val expectedMin = UInt.MAX_VALUE
+        val expectedMax = UInt.MAX_VALUE
+        val tokens = tokenizer.tokenize("$expectedMin funcref", context)
+        val parseResult = tokens.parseTableType(0)
+        assertThat(parseResult.astNode.limits.min).isEqualTo(expectedMin)
+        assertThat(parseResult.astNode.limits.max).isEqualTo(expectedMax)
+        assertThat(parseResult.parseLength).isEqualTo(2)
+    }
+
+    @Test
+    fun parseMinVal_setMinToMinVal() {
+        val expectedMin = UInt.MIN_VALUE
+        val expectedMax = UInt.MAX_VALUE
+        val tokens = tokenizer.tokenize("$expectedMin funcref", context)
+        val parseResult = tokens.parseTableType(0)
+        assertThat(parseResult.astNode.limits.min).isEqualTo(expectedMin)
+        assertThat(parseResult.astNode.limits.max).isEqualTo(expectedMax)
+        assertThat(parseResult.parseLength).isEqualTo(2)
+    }
+
+    @Test
+    fun parseNegativeMin_throwsParseExceptionWithNegativeNumberMessage() {
+        val tokens = tokenizer.tokenize("-123456", context)
+        val exception = assertThrows(ParseException::class.java) { tokens.parseTableType(0) }
+        assertThat(exception).hasMessageThat().contains("Expected integer literal")
```

Comment:
-123456 *is* an integer literal, so the error message here would be better as something like "Expected **unsigned** integer literal"